src/AxonIQ.AxonServer.Connector/AxonServerConnection.cs

```diff
@@ -129,19 +125,22 @@ private async Task RunChannelProtocol(CancellationToken ct)
                             switch (CurrentState)
                             {
                                 case State.Disconnected:
-                                    var channel = await _channelFactory.Create(_context);
+                                    var channel = await _channelFactory.Create(_context).ConfigureAwait(false);
                                     if (channel != null)
                                     {
-                                        var callInvoker = channel.CreateCallInvoker().Intercept(metadata =>
-                                        {
-                                            _channelFactory.Authentication.WriteTo(metadata);
-                                            _context.WriteTo(metadata);
-                                            return metadata;
-                                        });
+                                        var callInvoker = channel
+                                            .CreateCallInvoker()
+                                            .Intercept(_interceptors.ToArray())
```

Comment:
Order is important. The interceptors are registered in the order they are specified. The `Intercept` calls are however in "reverse" order, meaning that the last call to `Intercept` will be the first invoked in the pipeline. By registering the addition of authentication and context as the last call we make sure these are always set (even if subsequent interceptors could manipulate the metadata).